types/node/index.d.ts

```diff
@@ -1013,36 +1014,35 @@ interface IterableIterator<T> { }
 *                                               *
 *-----------------------------------------------*/
 declare module "buffer" {
-    export var INSPECT_MAX_BYTES: number;
-    var BuffType: typeof Buffer;
-    var SlowBuffType: typeof SlowBuffer;
+    export const INSPECT_MAX_BYTES: number;
+    const BuffType: typeof Buffer;
+    const SlowBuffType: typeof SlowBuffer;
     export { BuffType as Buffer, SlowBuffType as SlowBuffer };
 }
 
 declare module "querystring" {
-    export interface StringifyOptions {
+    interface StringifyOptions {
         encodeURIComponent?: Function;
     }
 
-    export interface ParseOptions {
+    interface ParseOptions {
         maxKeys?: number;
         decodeURIComponent?: Function;
     }
 
-    interface ParsedUrlQuery { [key: string]: string | string[] | undefined; }
+    interface ParsedUrlQuery { [key: string]: string | string[]; }
```

Comment:
The reason why `| undefined` was used here is that this allows you to augment `ParsedUrlQuery` like this in your application to easier access properties:
```interface ParsedUrlQuery {
  my_key?: string,
  my_arr_key?: string[]
}
```
if `| undefined` is not there it's only possible to specify mandatory properties.

Reply:
```
        let result: querystring.ParsedUrlQuery;

        const a: {
            a?: string;
        } = result;
```

works just fine in this case (pulled from test).

If you are talking about augmenting the interface through declaration merging however:
I think it's a really really bad practise to do that as you can have various definitions of query string parameters for all sorts of purposes, you should cast to your desired type instead.

Reply:
See the `http.IncomingHttpHeaders` which holds an index signature on the one hand and named optional properties on the other hand to narrow types for known/specified headers. This works only if `| undefined` is added in the index signature.

Users could do the same by augmenting `ParsedUrlQuery` but I agree that this is different and not really an intended usecase.